v1/bank/bank.gunk

```diff
@@ -0,0 +1,355 @@
+//                         Description: "Returned when the user does not have permission to access the resource.",
+//                 },
+//                 "404": openapiv2.Response{
+//                         Description: "Returned when the resource is not found.",
+//                 },
+//                 "500": openapiv2.Response{
+//                         Description: "Returned when an unexpected error occured on the server side.",
+//                 },
+//         },
+//         SecurityDefinitions: openapiv2.SecurityDefinitions{
+//                 Security: map[string]openapiv2.SecurityScheme{
+//                         "OAuth2": openapiv2.SecurityScheme{
+//                                 Type:             openapiv2.TYPE_OAUTH2,
+//                                 Flow:             openapiv2.FLOW_ACCESS_CODE,
+//                                 Description:      "Access to transactions resources is restricted to authenticated users. The authentication process follows the Authorization Code flow.",
+//                                 AuthorizationURL: "/v1/auth",
+//                                 TokenURL:         "/v1/token",
+//                                 Scopes: openapiv2.Scopes{
+//                                         Scope: map[string]string{
+//                                                 "read":  "Grants read access",
+//                                                 "write": "Grants write access",
+//                                         },
+//                                 },
+//                         },
+//                 },
+//         },
+// }
+package bank
+
+import (
+	"github.com/gunk/opt/http"
+	"github.com/gunk/opt/openapiv2"
+
+	types "github.com/openbank/openbank/v1/types"
+)
+
+// BankRouting holds all the details about a bank routing.
+type BankRouting struct {
+	// Scheme is the routing scheme.
+	Scheme string `pb:"1" json:"scheme"`
+
+	// Address is the routing address.
+	Address string `pb:"2" json:"address"`
+}
+
+// Bank holds all details about a bank.
+type Bank struct {
+	// ID is an identifier for the bank.
+	ID string `pb:"1" json:"id"`
+
+	// FullName is the full name of the bank.
+	FullName string `pb:"2" json:"full_name"`
+
+	// ShortName is the short name of the bank.
+	ShortName string `pb:"3" json:"short_name"`
+
+	// LogoURL is the url for the bank's logo.
+	LogoURL string `pb:"4" json:"logo_url"`
+
+	// WebsiteURL is the url for the bank's website.
+	WebsiteURL string `pb:"5" json:"website_url"`
+
+	// SwiftBIC is the SWIFT bank identifier code.
+	SwiftBIC string `pb:"6" json:"swift_bic"`
+
+	// NationalIdentifier is the national identifier code.
+	NationalIdentifier string `pb:"7" json:"national_identifier"`
+
+	// BankRouting is the routing information for the bank.
+	BankRouting BankRouting `pb:"8" json:"bank_routing"`
+}
+
+// TransactionType is holds the information about transaction type at a bank.
+type TransactionType struct {
+}
```

Comment:
why this struct is empty?

Reply:
good catch, I forgot to put the payload on this. 🙈